extension/mediator.js

```diff
@@ -338,7 +345,12 @@ class Mediator {
                     this.translate(message)
                     break
                 case "translationComplete":
-                    this.updateElements(message.translationMessage);
+                    if (getOrigin() === message.translationMessage.origin) {
+                        this.updateElements(message.translationMessage);
+                    } else {
+                        console.warn(`Message with a different origin is recieved by a frame. 
```

Comment:
Just for clarity: this issue is not specific to frames. A top-level navigation can also have the same issue, because tabId doesn't change and frameId will still be `0`.

Reply:
I added an origin check before updating HTML content for all "translationComplete" messages, not only frame ones